src/homework/client/src/actions/items-entities-actions.js

```diff
@@ -0,0 +1,101 @@
+import actionTypes from "./constants";
+import todoService from "../services/todo";
+import {addPokemonImage} from "../components/Main";
+
+
+
+// const updateSearchInput = (searchInput) => ({
+//   type: actionTypes.UPDATE_SEARCH_INPUT,
+//   payload: searchInput,
+// });
+
+// export const updateSearchInputAction = (searchInput) => {
+//   return async (dispatch) => {
+//     dispatch(updateSearchInput(searchInput));
+//   };
+// };
+
+  const updateDate= (object) => ({
+  type: actionTypes.UPDATE_DATE,
+  itemId: object.itemId,
+ // payload: object.date
+ });
+
+ export const updateDateAction= (object) => {
+  return async (dispatch) => {
+  await todoService.updateDate(object);
+  dispatch(updateDate(object));
+  };
+ };
+
+ export const getSortAction = () => {
+   return async (dispatch) => {
+    const items = await todoService.getSorted();
+     dispatch(addItems(items));
+   };
+ };
+
+const clearAllItems = () => ({
+  type: actionTypes.CLEAR_ALL_ITEMS,
+});
+
+export const clearAllItemsAction = () => {
+  
+  return async (dispatch) => {
+    await todoService.removeAll()
+    dispatch(clearAllItems());
+  };
+};
+
+const addItems = (newItems) => ({
+  type: actionTypes.ADD_ITEMS,
+  payload: newItems,
+});
+
+export const addItemsAction = (newItems) => {
+  return async (dispatch) => {
+    const addedItems = await todoService.create(newItems);
+    if(addedItems.error){
```

Comment:
for errors handling it's better to use relevant status codes in the server (200 for success, 500 for internal server error... you can read more. search for HTTP response status code.
This way you can add callback that will be triggered in case of failure

<img width="424" alt="image" src="https://user-images.githubusercontent.com/73115367/178934615-8e38280a-4c34-42fd-af42-4b2289179ba6.png">